specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json

```diff
@@ -0,0 +1,1350 @@
+        },
+        "protocol": {
+          "$ref": "#/definitions/AzureFirewallNetworkRuleProtocol",
+          "description": "The protocol of packets to capture"
+        },
+        "flags": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AzureFirewallPacketCaptureFlags"
+          },
+          "description": "The tcp-flag type to be captured. Used with protocol TCP",
+          "x-ms-identifiers": []
+        },
+        "filters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AzureFirewallPacketCaptureRule"
+          },
+          "description": "Rules to filter packet captures.",
+          "x-ms-identifiers": []
+        }
+      },
+      "description": "Azure Firewall Packet Capture Parameters."
+    },
+    "AzureFirewallPacketCaptureFlags": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Flags to capture",
+          "$ref": "#/definitions/AzureFirewallPacketCaptureFlagsType"
+        }
+      },
+      "description": "Properties of the AzureFirewallRCAction."
+    },
+    "AzureFirewallPacketCaptureFlagsType": {
+      "type": "string",
+      "description": "The flags type to be captured.",
+      "enum": [
+        "fin",
+        "syn",
+        "rst",
+        "push",
+        "ack",
+        "urg"
+      ],
+      "x-ms-enum": {
+        "name": "AzureFirewallPacketCaptureFlagsType",
+        "modelAsString": true
+      }
+    },
+    "AzureFirewallPacketCaptureRule": {
+      "type": "object",
+      "properties": {
+        "sources": {
+          "type": "array",
+          "description": "List of source IP addresses/subnets to be captured.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "destinations": {
+          "type": "array",
+          "description": "List of destination IP addresses/subnets to be captured.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "destinationPorts": {
+          "type": "array",
+          "description": "List of ports to be captured.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "description": "Group of src/dest ips and ports to be captured."
+    },
+    "AzureFirewallAutoscaleConfiguration": {
+      "type": "object",
+      "properties": {
+        "minCapacity": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 2,
+          "description": "The minimum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+          "x-nullable": true
+        },
+        "maxCapacity": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 2,
+          "description": "The maximum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+          "x-nullable": true
```

Comment:
>nullable [](http://example.com/codeflow?start=13&length=8)

x-nullable, or representation with explicit nulls, is not generally recommended in Azure APIs.
Is there a reason you need it??